### PR TITLE
Status reporting

### PR DESCRIPTION
**Pull request by cmdcolin**
I felt sufficiently disturbed by https://github.com/GMOD/jbrowse-components/pull/1142 that I decided to see what it might look like to add status reporting to this cache system

I thought it didn't seem too bad and resulted in something that was sort of similar to the aggregate abort concept

Not yet tested in the context of jb2 status reporting

### CODE REVIEW

File: src/AbortablePromiseCache.ts [hunk 1]
```diff
@@ -69,12 +75,17 @@ export default class AbortablePromiseCache<T, U> {
     if (this.cache.get(key) === entry) this.cache.delete(key)
   }
 
-  fill(key: string, data: T, signal?: AbortSignal) {
+  fill(key: string, data: T, signal?: AbortSignal, statusCallback?: Function) {
     const aborter = new AggregateAbortController()
+    const statusReporter = new AggregateStatusReporter()
+    statusReporter.addCallback(statusCallback)
```
Comment (rbuels) [inline src/AbortablePromiseCache.ts:81]: What happens here if statusCallback is undefined?

Comment (cmdcolin) [reply src/AbortablePromiseCache.ts:81]: the AggregateStatusReporter has a default argument assignment to empty function. benefit of typescripting the module...makes sure things like this are ok

**Comment by rbuels** [comment]
Should the status callback be called with an "Aborted" notification when things are aborted, I wonder? I kind of think so.

File: src/AbortablePromiseCache.ts [hunk 2]
```diff
@@ -139,7 +150,12 @@ export default class AbortablePromiseCache<T, U> {
    * @param {any} data data passed as the first argument to the fill callback
    * @param {AbortSignal} [signal] optional AbortSignal object that aborts the request
```
Comment (rbuels) [inline src/AbortablePromiseCache.ts:158]: need to add statusCallback to the jsdoc here

**Review by rbuels — APPROVED** [review]
looks good, just the one little piece of jsdoc still needed

File: src/AbortablePromiseCache.ts (hunk 2 above)
Comment (cmdcolin) [reply src/AbortablePromiseCache.ts:158]: added the jsdoc with the full callback syntax :)

**Comment by cmdcolin** [comment]
not sure I would report the aborted to status or not. it seems like the requester will get an aborted exception if something does happen? so then user code would handle that and show that aborted exception to the end user if needed?

**Comment by rbuels** [comment]
yeah that's true. ok going to merge and publish

**Comment by cmdcolin** [comment]
Try changing the postpublish to postversion if you want to try the auto-push-to-github-after-publish too :)

**Comment by rbuels** [comment]
alright published as 1.3.0

**Pull request merged**